arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1428,16 +1458,24 @@ Result RocksDBCollection::updateDocument(transaction::Methods* trx,
   }
 
   READ_LOCKER(guard, _indexesLock);
-  for (std::shared_ptr<Index> const& idx : _indexes) {
-    RocksDBIndex* rIdx = static_cast<RocksDBIndex*>(idx.get());
-    Result tmpres = rIdx->updateInternal(trx, mthd, oldDocumentId, oldDoc, newDocumentId,
+  bool needReversal = false;
+  for (auto it = _indexes.begin(); it != _indexes.end(); it++) {
+    auto rIdx = static_cast<RocksDBIndex*>(it->get());
+    res = rIdx->updateInternal(trx, mthd, oldDocumentId, oldDoc, newDocumentId,
                                          newDoc, options.indexOperationMode);
-    if (!tmpres.ok()) {
-      if (tmpres.is(TRI_ERROR_OUT_OF_MEMORY)) {
+    needReversal = needReversal || rIdx->needsReversal();
+    if (!res.ok()) {
+      if (res.is(TRI_ERROR_OUT_OF_MEMORY)) {
```

Comment:
I think we should also get rid of the special handling for out of memory here.
Because if it is triggered, we will return immediately, and no rollback will happen.
So let's get rid of any special handling for any error code here.